### PR TITLE
Update User&URL Response Format to JSON API Standard

### PR DESCRIPTION
**Pull request by vinit717**
### **Description:**
The current response format for user/URL-related endpoints does not adhere to the JSON API standard, leading to inconsistencies and potential difficulties for client-side consumption.

**Changes Made:**
Modified the User struct in the models package to include json tags for each field, following the JSON API standard.
Omitted the Password field from JSON responses for security reasons.

**Proof:**
image
image

### CODE REVIEW

**Review by sahsisunny — APPROVED** [review]

File: controllers/url.go [hunk 1]
```diff
@@ -111,4 +111,4 @@ func GetURLDetails(ctx *gin.Context, db *bun.DB) {
 		"message": "URL fetched successfully",
 		"url":     tinyURL,
 	})
-}
\ No newline at end of file
+}
```
Comment (shreya-mishra) [inline controllers/url.go:114]: why this change ?

Comment (vinit717) [reply controllers/url.go:114]: UNIX standard

**Review by shreya-mishra — APPROVED** [review]
